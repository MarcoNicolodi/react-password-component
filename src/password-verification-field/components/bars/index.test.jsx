import React from "react";
import renderer from "react-test-renderer";
import Bars from "./";

describe("Bars component", () => {
  it("should match snapshot when password field is empty", () => {
    const tree = renderer.create(<Bars strength={null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot when password is weak", () => {
    const tree = renderer.create(<Bars strength="weak" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot when password is strong", () => {
    const tree = renderer.create(<Bars strength="strong" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot when password is very strong", () => {
    const tree = renderer.create(<Bars strength="very-strong" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
