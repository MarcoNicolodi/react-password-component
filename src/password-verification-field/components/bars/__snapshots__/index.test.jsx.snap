// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bars component should match snapshot when password field is empty 1`] = `
<div
  className="bars"
>
  <div
    className="bar "
  />
  <div
    className="bar "
  />
  <div
    className="bar "
  />
</div>
`;

exports[`Bars component should match snapshot when password is strong 1`] = `
<div
  className="bars"
>
  <div
    className="bar strong"
  />
  <div
    className="bar strong"
  />
  <div
    className="bar "
  />
</div>
`;

exports[`Bars component should match snapshot when password is very strong 1`] = `
<div
  className="bars"
>
  <div
    className="bar very-strong"
  />
  <div
    className="bar very-strong"
  />
  <div
    className="bar very-strong"
  />
</div>
`;

exports[`Bars component should match snapshot when password is weak 1`] = `
<div
  className="bars"
>
  <div
    className="bar weak"
  />
  <div
    className="bar "
  />
  <div
    className="bar "
  />
</div>
`;
